Open source publish

silverjam opened the pull request:
Open source publish of df25faf9f28eb74ace9f9765c451087320b2d8ba..cae2886bb7ef4baa3c44864b5b637ea3d227a25f

Jira: https://swift-nav.atlassian.net/browse/DIP-307

- Install the leap seconds header
- updated leap second
- bazel: add swift_cc* wrappers
- bazel: fix code-coverage
- Remove bazel submodule
  - Fetches swift's bazel rules via http_archive instead of using a submodule
- Add cn0 mask to calc-pvt
- bazel: Require build passes in CI
- Export decode_sisa_index(), decode_bds_ura_index() and decode_fit_interval() functions: This allows external code to reuse libswiftnav's functions for decoding URA/SISA index values.
  - Also includes a couple of minor improvements:
    - Move definition of GAL_WEEK_TO_GPS_WEEK from ephemeris.h to gnss_time.h to be consistent with BDS_WEEK_TO_GPS_WEEK
    - Fix test for platforms where time_t is a signed 32-bit integer
- Added Makefile with bazel wrappers
- Fix group delay type for B1C
- bazel: fix optimized compilation
- clean up iono handling and navigation_measurement_t
- bazel add gen_compile_commands target
- bazel: make max_channels a configurable attribute
- bazel: use c only macros to build libswiftnav
- Handle endianess in bytestreams
- Byte swap inline function
- Regenerate leap second
- bazel: Add bazeliskrc file
- Use float whenever possible in satellite attitude and std
- Revert "Use float whenever possible in satellite attitude and std"
- Use float is satellite vel and acc when possible: Changed satellite acceleration and velocity to float when possible
- Handle BDS B3I in get_tgd_correction()
- Revert "Use float is satellite vel and acc when possible"
- Add support for GPS L1CQ in health check

---

Co-authored-by: Jason Mobarak <jason@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Altti Jokinen <altti.jokinen@swift-nav.com>
Co-authored-by: Marco <marco.mendonca@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Antriksh Srivastava <antriksh.srivastava@swift-nav.com>
Co-authored-by: Joseph Angelo <joseph.angelo@swift-nav.com>
Co-authored-by: Isaac Torres <isaac.torres@swift-nav.com>
Co-authored-by: Lucas Le <97814268+lucasle-sn@users.noreply.github.com>
Co-authored-by: Rodrigo Reichert <rodrigo.reichert@swift-nav.com>
Co-authored-by: Matt Woodward <46688854+woodfell@users.noreply.github.com>
Co-authored-by: Wiktor Więcław <110531204+wwieclaw@users.noreply.github.com>
Co-authored-by: Krzysztof Naglik <krzysztofnaglik96@gmail.com>
Co-authored-by: dgburr <dburr@dburr.net>

mbavaro commented:
@martin-swift does this hurt your feelings?

martin-swift commented:
Might be totally reasonable, except what problem is this solving?

mbavaro commented:
I just need something to associate a `code_t` to a string, `sat_code_to_string()` does not cut it, nor `sid_to_string()` or `mesid_to_string()`.

mbavaro commented:
Thanks Martin, mergin away

jbangelo committed:
Open source publish df25faf9f28eb74ace9f9765c451087320b2d8ba..cae2886bb7ef4baa3c44864b5b637ea3d227a25f

- Install the leap seconds header
- updated leap second
- bazel: add swift_cc* wrappers
- bazel: fix code-coverage
- Remove bazel submodule
  - Fetches swift's bazel rules via http_archive instead of using a submodule
- Add cn0 mask to calc-pvt
- bazel: Require build passes in CI
- Export decode_sisa_index(), decode_bds_ura_index() and decode_fit_interval() functions: This allows external code to reuse libswiftnav's functions for decoding URA/SISA index values.
  - Also includes a couple of minor improvements:
    - Move definition of GAL_WEEK_TO_GPS_WEEK from ephemeris.h to gnss_time.h to be consistent with BDS_WEEK_TO_GPS_WEEK
    - Fix test for platforms where time_t is a signed 32-bit integer
- Added Makefile with bazel wrappers
- Fix group delay type for B1C
- bazel: fix optimized compilation
- clean up iono handling and navigation_measurement_t
- bazel add gen_compile_commands target
- bazel: make max_channels a configurable attribute
- bazel: use c only macros to build libswiftnav
- Handle endianess in bytestreams
- Byte swap inline function
- Regenerate leap second
- bazel: Add bazeliskrc file
- Use float whenever possible in satellite attitude and std
- Revert "Use float whenever possible in satellite attitude and std"
- Use float is satellite vel and acc when possible: Changed satellite acceleration and velocity to float when possible
- Handle BDS B3I in get_tgd_correction()
- Revert "Use float is satellite vel and acc when possible"
- Add support for GPS L1CQ in health check

---

Co-authored-by: Jason Mobarak <jason@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Altti Jokinen <altti.jokinen@swift-nav.com>
Co-authored-by: Marco <marco.mendonca@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Antriksh Srivastava <antriksh.srivastava@swift-nav.com>
Co-authored-by: Joseph Angelo <joseph.angelo@swift-nav.com>
Co-authored-by: Isaac Torres <isaac.torres@swift-nav.com>
Co-authored-by: Lucas Le <97814268+lucasle-sn@users.noreply.github.com>
Co-authored-by: Rodrigo Reichert <rodrigo.reichert@swift-nav.com>
Co-authored-by: Matt Woodward <46688854+woodfell@users.noreply.github.com>
Co-authored-by: Wiktor Więcław <110531204+wwieclaw@users.noreply.github.com>
Co-authored-by: Krzysztof Naglik <krzysztofnaglik96@gmail.com>
Co-authored-by: dgburr <dburr@dburr.net>

silverjam committed:
bazel?

silverjam committed:
more bazel updates

silverjam commented:
Not sure how bazel merge got messed up, but manually copied over changes and diff'd things to make sure I got everything:

```
❯ diff -x '.git' -ru libswiftnav-private libswiftnav
diff '--color=auto' -x .git -ru libswiftnav-private/ci-build.sh libswiftnav/ci-build.sh
--- libswiftnav-private/ci-build.sh	2022-11-09 15:37:58.782657989 -0800
+++ libswiftnav/ci-build.sh	2023-09-29 16:34:20.625853099 -0700
@@ -93,6 +93,8 @@
         -Dsonar.cfamily.cache.enabled=false \
         -Dsonar.sourceEncoding=UTF-8"
 
+    set -x
+
     # shellcheck disable=SC2086
     sonar-scanner \
         "-Dsonar.cfamily.llvm-cov.reportPath=./build/ccov/coverage.txt" \
diff '--color=auto' -x .git -ru libswiftnav-private/CMakeLists.txt libswiftnav/CMakeLists.txt
--- libswiftnav-private/CMakeLists.txt	2023-04-27 15:17:12.581893870 -0700
+++ libswiftnav/CMakeLists.txt	2023-09-29 16:45:22.666112595 -0700
@@ -134,4 +134,4 @@
 )
 
 swift_validate_targets()
-swift_create_clang_tidy_targets()
+swift_create_clang_tidy_targets(DONT_GENERATE_CLANG_TIDY_CONFIG)
diff '--color=auto' -x .git -ru libswiftnav-private/README.md libswiftnav/README.md
--- libswiftnav-private/README.md	2022-11-09 15:37:58.782657989 -0800
+++ libswiftnav/README.md	2023-09-29 16:34:20.625853099 -0700
@@ -1,8 +1,9 @@
-[![Build Status](https://jenkins-webhooks.ci.swift-nav.com/buildStatus/icon?job=swift-nav/libswiftnav-private/master&subject=Jenkins&style=flat)](https://jenkins.ci.swift-nav.com/job/swift-nav/job/libswiftnav-private/job/master/) [![CI](https://github.com/swift-nav/libswiftnav-private/actions/workflows/ci.yaml/badge.svg)](https://github.com/swift-nav/libswiftnav-private/actions/workflows/ci.yaml)
+[![CI](https://github.com/swift-nav/libswiftnav/actions/workflows/ci.yaml/badge.svg)](https://github.com/swift-nav/libswiftnav/actions/workflows/ci.yaml)
+[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=swift-nav_libswiftnav&metric=alert_status)](https://sonarcloud.io/dashboard?id=swift-nav_libswiftnav)
 
 
-libswiftnav-private
-===================
+libswiftnav
+===========
 
 Libswiftnav (LSN) is a platform independent library that implements GNSS utility functions for use by software-defined GNSS receivers or software requiring GNSS functionality. It is intended to be as portable as possible and is written in standards compliant C with no dependancies
 
@@ -10,7 +11,7 @@
 
 To checkout the library run the following commands in an appropriate directory
 ```
-git clone git@github.com:swift-nav/libswiftnav-private.git
+git clone git@github.com:swift-nav/libswiftnav.git
 ```
 Which should checkout the source code
```

silverjam committed:
bump cmake
